middleware/authentication.js

```diff
@@ -0,0 +1,23 @@
+const User = require('../models/User')
```

Comment:
[Functional Testing]
So this code looks right, but it wasn't tested. I know cause its called anywhere and these references I called out would've blown up.

The user controller is being added in another PR that is almost done. Lets get that in and then call this middleware on the get user by id route. Then test via Postman.

Reply:
I will try to test using the get by Id, which I haven't yet, only registering and login so far..